docs/agent-policy.md

```diff
@@ -0,0 +1,169 @@
+# Elastic Agent Policies
+
+The policy of an Agent is the user-controlled portion of the Agent configuration that, together with [component spec files](component-specs.md), determines the components that Agent runs and their configuration.
+
+A __component__ is an executable that Agent runs or monitors. That component is divided into __units__ corresponding to its functional behavior. For example, Filebeat is a component, but Filebeat may be run with multiple `filestream` inputs, and each of those is a unit with its own configuration.
+
+Components run by Agent have one or more input units describing the data they collect, and one output unit describing where the data goes.
```

Comment:
Might be helpful to link back to the diagram in https://github.com/elastic/elastic-agent/blob/main/ARCHITECTURE.md#compute-model

It might also make sense for that architecture file to be moved into this directory with all of this documentation linked from the readme. We currently only link to the architecture file. https://github.com/elastic/elastic-agent/blob/main/README.md#architecture

Reply:
👍 moved the architecture doc into `docs`, linked it from here, and also linked the new docs from the readme.